docs/proposals/multiple-schedulers.md

```diff
@@ -0,0 +1,139 @@
+Kubernetes native scheduler (now `kube-scheduler` process) would have the name as `kube-scheduler`
+- The `scheduler-name` plays an important part in separating the pods between different schedulers
+which is done by carefully designing the condition/logic of how each scheduler selects unscheduled
+pods to add to its queue. To be specific, a scheduler can add a pod to its queue if and only if:
+    1. The pod has no host, **AND**
+    2. The `scheduler.alpha.kubernetes.io/scheduler-name` specified in the pod annotation matches
+    the scheduler name.
+
+        One exception is the default scheduler (with name `kube-scheduler`) who should be allowed
+        to add a none host pod to its queue if: a.) the pod DOES NOT contain annotation with key
+        `scheduler.alpha.kubernetes.io` **OR** b.) the pod contains annotation `scheduler.alpha.kubernetes.io/kube-scheduler`
+
+- Dealing with conflicts. All schedulers must use predicate functions that are at least as strict as
+the ones that Kubelet applies when deciding whether to accept a pod, otherwise Kubelet and scheduler
+may get into an infinite loop where Kubelet keeps rejecting a pod and scheduler keeps re-scheduling
+it back the same node. To make it easier for people who write new schedulers to obey this rule, we will
+create a library containing the predicates Kubelet uses. (See issue [#12744](https://github.com/kubernetes/kubernetes/issues/12744).)
+
+In summary, in the initial version of this multi-scheduler design, we will achieve the following:
+
+- If a pod has the annotation `scheduler.alpha.kubernetes.io/kube-scheduler` or the user does not explicitly
+sets this annotation in the template, it will be picked up by default scheduler
+- If the annotation is set and refers to a valid `scheduler-name`, it will be picked up by the scheduler of
+specified `scheduler-name`
+- If the annotation is set but refers to an invalid `scheduler-name`, the pod will not be picked by any scheduler.
+Since there is no admission controller or other thing alike to verify the annotation, there will be no error or
+warning for the user. The pod will keep PENDING.
+
+### An example
+
+```
+    kind: Pod
+    apiVersion: v1
+    metadata:
+        name: pod-abc   
+        labels:
+            foo: bar
+        annotations:
+            scheduler.alpha.kubernetes.io: my-scheduler
+```
+
+This pod will be scheduled by "my-scheduler" and ignored by "kube-scheduler". If there is no running scheduler
+of name "my-scheduler", the pod will never be scheduled.
+
+## Next steps
+
+1. Use admission controller to verify the annotation, and do some modification if necessary. For example, the
+admission controller might add the scheduler annotation based on the namespace of the pod, and/or identify if
+there are conflicting rules, and/or set a default value for the scheduler annotation, and/or reject pods on
+which the client has set a scheduler annotation that does not correspond to a running scheduler.
```

Comment:
This doc isn't really intended to design how the annotations get added (but it's useful to at least give a flavor as to the kinds of policies we might have). These are good comments but I think we should leave this doc at the current level of detail, and address these comments in a follow-up design for how to set the annotations.